lisa/analysis/rta.py

```diff
@@ -209,6 +318,56 @@ def tasks_window(self, task):
 
         return (start_time, end_time)
 
+    @df_rtapp_phases_start.used_events
+    def task_phase_window(self, task, phase):
+        """
+        Return the window of a requested task phase.
+
+        For the specified ``task`` it returns a tuple with the (start, end)
+        time of the requested ``phase``. A negative phase number can be used to
+        count phases backward from the last (-1) toward the first.
+
+        :param task: the (optional) rt-app task to filter for
+        :type task: int or str or :class:`TaskID`
+
+        :param phase: The ID of the phase to consider
+        :type phase: int
+
+        :returns: tuple(start_s, end_s)
+        """
+        phase_start = self.df_rtapp_phase_start(task, phase)
+        phase_end = self.df_rtapp_phase_end(task, phase)
+
+        return self.PhaseWindow(phase, phase_start, phase_end)
+
+    @task_phase_window.used_events
+    def task_phase_at(self, task, timestamp):
+        """
+        Return the :class:`PhaseWindow` for the specified task and timestamp.
+
+        :param task: the (optional) rt-app task to filter for
+        :type task: int or str or :class:`TaskID`
+
+        :param timestamp: the timestamp to get the pahse for
+        :type timestamp: int or float
+
+        :returns: the ID of the phase corresponding to the specified timestamp,
+                  None if the specified timestamp does not match a phase
+        """
+        # Last phase is special, compute end time as start + duration
+        last_phase_end = self.df_phases('test_task-0').index[-1]
+        last_phase_end += float(self.df_phases('test_task-0').iloc[-1].values)
+        if timestamp > last_phase_end:
```

Comment:
When something unexpected like that happens, it's a better idea to raise an exception. If caller code can cope with that, it will catch it. APIs based on returning `None` lead to "random" attribute errors and sprinkling checks of `None` like `if foo is None: return None`, which is exactly what exception will provide without the noise in the code.